storage/directory.go

```diff
@@ -121,23 +121,80 @@ func (s *DirectoryClient) Put(ctx context.Context, input *PutDirectoryInput) err
 // DeleteDirectoryInput represents parameters to a Delete operation.
 type DeleteDirectoryInput struct {
 	DirectoryName string
+	ForceDelete   bool
```

Comment:
Can we comment what this does in the `struct` definition and call out that it will recursively delete both objects and files?